modules/dtj/Animal/scripts/barnyard.pl

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+
+use Cow;
+use Horse;
+use Sheep;
+use Mouse;
+
+my @animals;
+print "Enter some Animals to go into the barnyard, one per line\n";
+print "Press ctrl+d when finished\n";
+while ( <> ) {
+    chomp;
+    my $input = $_;
+
+    push @animals, $input;
+}
+
+foreach my $animal ( @animals ) {
+    eval {
+        $animal->speak;
```

Comment:
I'm fairly sure animal is a string. It's documented behaviour [here](http://perldoc.perl.org/perlobj.html#Method-Call-Variations) that you can have a String containing a class name and call methods from it - as weird and wonderful as that sounds.